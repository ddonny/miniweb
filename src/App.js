import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './App.scss';
import {
  Row,
  Col
} from 'react-materialize'
/* components */
import Navbar from 'components/Navbar';
import Body from 'components/Body';
import LeftMenu from 'components/LeftMenu';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Navbar />
        <Row className="body-content">
          <Col s={12} m={3} l={3} className="is-side">
            <LeftMenu />
          </Col>
          <Col s={12} m={9} l={9} className="is-main">
            <Body />
          </Col>
        </Row>     
      </div>
    );
  }
}

const mapStateToProps = state => ({
  ...state
})

const mapDispatchToProps = dispatch => ({
})

App.propTypes = {
  globalReducer: PropTypes.any
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
