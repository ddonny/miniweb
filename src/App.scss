/* 
  ##Device = Desktops
  ##Screen = 1281px to higher resolution desktops
*/
// @media (min-width: 1281px) {}
/* 
  ##Device = Laptops, Desktops
  ##Screen = B/w 1025px to 1280px
*/
// @media (min-width: 1025px) and (max-width: 1280px) {}
/* 
  ##Device = Tablets, Ipads (portrait)
  ##Screen = B/w 768px to 1024px
*/
// @media (min-width: 768px) and (max-width: 1024px) {}
/* 
  ##Device = Tablets, Ipads (landscape)
  ##Screen = B/w 768px to 1024px
*/
// @media (min-width: 768px) and (max-width: 1024px) and (orientation: landscape) {}
/* 
  ##Device = Low Resolution Tablets, Mobiles (Landscape)
  ##Screen = B/w 481px to 767px
*/
// @media (min-width: 481px) and (max-width: 767px) {}
/* 
  ##Device = Most of the Smartphones Mobiles (Portrait)
  ##Screen = B/w 320px to 479px
*/
// @media (min-width: 320px) and (max-width: 480px) {}
@import 'components/Navbar/index.scss';
@import 'components/Body/index.scss';
@import 'components/LeftMenu/index.scss';
input:not([type]):focus:not([readonly])+label, input[type=text]:focus:not([readonly])+label, input[type=password]:focus:not([readonly])+label, input[type=email]:focus:not([readonly])+label, input[type=url]:focus:not([readonly])+label, input[type=time]:focus:not([readonly])+label, input[type=date]:focus:not([readonly])+label, input[type=datetime]:focus:not([readonly])+label, input[type=datetime-local]:focus:not([readonly])+label, input[type=tel]:focus:not([readonly])+label, input[type=number]:focus:not([readonly])+label, input[type=search]:focus:not([readonly])+label, textarea.materialize-textarea:focus:not([readonly])+label{
  color: #9e9e9e;
}
body{
  background-color: #E5E5E8;
  font-size: 16px;
}
.row{
  margin-bottom: 0;
}
.flex{
  display: flex;
}
.align-center{
  text-align: center;
}