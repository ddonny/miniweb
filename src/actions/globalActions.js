export const setSelectedMenu = (payload) => {
  return {
    type: 'SET_SELECTED_MENU',
    selectedFromListMenu: payload
  }
}
export const setInputNameValue = (payload) => {
  return {
    type: 'SET_INPUT_VALUE',
    inputNameValue: payload
  }
}
export const setPostData = (payload) => {
  return {
    type: 'SET_POST_DATA',
    postData: payload
  }
}

