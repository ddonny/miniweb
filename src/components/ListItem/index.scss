.list-item{
  h4{
    padding: 0 30px 0 30px;
  }
  .section-wrapper{
    padding: 0 30px 30px 30px;
    table{
      td{
        &.ellipsis{
          text-overflow: ellipsis;
          white-space: nowrap;
          overflow: hidden;
          max-width: 10rem;
        }
      }
    }
    .pagination{
      text-align: center;
      li{
        &.active{
          background-color: #CCCCCC;
        }
      }
    }
  }
}