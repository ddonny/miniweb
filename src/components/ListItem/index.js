import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import 'components/ListItem/index.scss';
import {
  Table,
  Pagination
} from 'react-materialize';
import Loader from 'components/Loader';
import { setPostData} from 'actions/globalActions';

class ListItem extends Component {
  constructor(props){
    super(props);
    this.state = {
      requestComplete: (props.requestComplete) ? props.requestComplete : false,
      activePage: 1,
      chunkPerPage: 10,
      availablePage: 1,
      groupsPostData: (props.groupsPostData) ? props.groupsPostData : [],
      groupsPostDataByPage: (props.groupsPostDataByPage) ? props.groupsPostDataByPage : []
    }
  }
  componentDidMount() {
    const {
      requestComplete
    } = this.state;
    if (!requestComplete) {
      fetch("https://jsonplaceholder.typicode.com/posts")
      .then(res => res.json())
      .then(
        (result) => {
          this.processPostData(result);
        },
        (error) => {
          this.setState({
            error: error
          });
        }
      )
    }
  }
  wait(ms, data) {
    return new Promise(resolve => setTimeout(resolve.bind(this, data), ms));
  }
  async processPostData(postData) {
    const a = {
        act1: await this.wait(200, this.props.setPostData(postData)),
        act2: await this.wait(200,
            this.setData(postData)
        ),
        act3: await this.wait(200,
          this.setState({
            requestComplete: true
          })
        ),
        act4: await this.wait(200,
          this.getDataByCurrentPage()
        )
    };
    return a
  }
  splitArray = (array, chunk_size) => {
    var results = [];
    while (array.length) {
        results.push(array.splice(0, chunk_size));
    }  
    return results;
  }
  setData = (postData) => {
    const {
      activePage,
      chunkPerPage,
      availablePage
    } = this.state;
    let groups = this.splitArray(postData, chunkPerPage)
    this.setState({
      groupsPostData: groups,
      availablePage: groups.length
    });
  }
  getDataByCurrentPage = (page) => {
    const {
      activePage,
      chunkPerPage,
      availablePage,
      groupsPostData
    } = this.state;
    let currentPage;
    if (!page) {
      currentPage = activePage-1; 
    } else {
      currentPage = page-1;
    }
    this.setState({
      groupsPostDataByPage: groupsPostData[currentPage]
    })
  }
  onSelectPagination = (e) => {
    this.getDataByCurrentPage(e);
  }
  render() {
    const {
      requestComplete,
      error,
      activePage,
      chunkPerPage,
      availablePage,
      groupsPostDataByPage
    } = this.state;
    const {
      postData
    } = this.props.globalReducer;
    return (
      <div className="list-item">
        <h4>
          List of Items with pagination
        </h4>
          {
            (error) ?
            <div className="section-wrapper align-center">
              {
                JSON.stringify(error)
              }
            </div>
            :
            (!requestComplete || !postData || !groupsPostDataByPage) ?
            <div className="section-wrapper align-center">
              <Loader />
            </div>
            :
            <div className="section-wrapper">
              <Table>
                <thead>
                  <tr>
                    <th data-field="id">ID</th>
                    <th data-field="userId">User ID</th>
                    <th data-field="title">Title</th>
                  </tr>
                </thead>

                <tbody>
                  {
                    (groupsPostDataByPage).map((item, index) => 
                      <tr key={index}>
                        <td>{item.id}</td>
                        <td>{item.userId}</td>
                        <td className="ellipsis">{item.body}</td>
                      </tr>
                    )
                  }
                </tbody>
              </Table>
              <Pagination items={chunkPerPage} activePage={activePage} maxButtons={availablePage} onSelect={this.onSelectPagination}/>
            </div>
          }
        </div>
    );
  }
}

const mapStateToProps = state => ({
  ...state
})

const mapDispatchToProps = dispatch => ({
  setPostData: (payload) => dispatch(setPostData(payload))
})

ListItem.propTypes = {
  globalReducer: PropTypes.any,
  dispatch: PropTypes.any,
  setPostData: PropTypes.any,
  requestComplete: PropTypes.bool,
  groupsPostData: PropTypes.array,
  groupsPostDataByPage: PropTypes.array
}

export default connect(mapStateToProps, mapDispatchToProps)(ListItem);
