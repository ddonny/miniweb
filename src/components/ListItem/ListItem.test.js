import React from 'react';
import ReactDOM from 'react-dom';
import App from 'App';
import Enzyme, { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { setSelectedMenu, setInputNameValue, setPostData } from 'actions/globalActions';
import ListItem from 'components/ListItem/index';
import {
  Table,
  Pagination
} from 'react-materialize';
const middlewares = []
const mockStore = configureStore(middlewares)
const store = mockStore();
Enzyme.configure({ adapter: new Adapter() })
const Provider = require('react-redux').Provider;
const createStore = require('redux').createStore;
const props = {
  globalReducer: {
    selectedFromListMenu: 'home',
    inputNameValue: 'myname',
    postData: [
      {
        "userId": 1,
        "id": 1,
        "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
        "body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      }
    ],
    requestComplete: true,
    activePage: 1,
    chunkPerPage: 10,
    availablePage: 1,
    groupsPostDataByPage: [[
      {
        "userId": 1,
        "id": 1,
        "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
        "body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      },
      {
        "userId": 1,
        "id": 2,
        "title": "qui est esse",
        "body": "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla"
      }
    ]]
  }
};
describe('List Item test', function () {
  beforeEach(() => {
    jest.setTimeout(5000);
    jest.useFakeTimers();
  });
  it('List Item', (done) => {
    const store = mockStore(props)
    const context = { store };
    const wrapper = mount(
        <ListItem/>,
        { context },
      );
    let findText = wrapper.find('h4').text();
    expect(findText).toEqual('List of Items with pagination');
    done()
  })
  it('listitem match with snapshot', (done) => {
    const store = mockStore(props)
    const context = { store };
    const wrapper = shallow(
        <ListItem/>,
        { context },
      );
    wrapper.dive().instance().processPostData(props.globalReducer.postData);
    expect(wrapper).toMatchSnapshot();
    done()
  })
  it('listitem render table and pagination', (done) => {
    const store = mockStore(props)
    const context = { store };
    const wrapper = shallow(
        <ListItem groupsPostData={props.globalReducer.groupsPostDataByPage} groupsPostDataByPage={props.globalReducer.groupsPostDataByPage} requestComplete={props.globalReducer.requestComplete}/>,
        { context }
    ).dive({context});
    // console.log(wrapper.state())
    // console.log(wrapper.debug())
    expect(wrapper.find('Table')).toHaveLength(1);
    expect(wrapper.find('Pagination')).toHaveLength(1);
    done()
  })
});