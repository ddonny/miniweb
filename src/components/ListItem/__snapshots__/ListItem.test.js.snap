// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item test listitem match with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(ListItem) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "globalReducer": Object {
        "activePage": 1,
        "availablePage": 1,
        "chunkPerPage": 10,
        "groupsPostDataByPage": Array [
          Array [
            Object {
              "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
              "id": 1,
              "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
          ],
        ],
        "inputNameValue": "myname",
        "postData": Array [
          Object {
            "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
            "id": 1,
            "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
            "userId": 1,
          },
          Object {
            "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
            "id": 2,
            "title": "qui est esse",
            "userId": 1,
          },
          Object {
            "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
            "id": 2,
            "title": "qui est esse",
            "userId": 1,
          },
          Object {
            "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
            "id": 2,
            "title": "qui est esse",
            "userId": 1,
          },
          Object {
            "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
            "id": 2,
            "title": "qui est esse",
            "userId": 1,
          },
          Object {
            "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
            "id": 2,
            "title": "qui est esse",
            "userId": 1,
          },
          Object {
            "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
            "id": 2,
            "title": "qui est esse",
            "userId": 1,
          },
          Object {
            "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
            "id": 2,
            "title": "qui est esse",
            "userId": 1,
          },
          Object {
            "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
            "id": 2,
            "title": "qui est esse",
            "userId": 1,
          },
          Object {
            "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
            "id": 2,
            "title": "qui est esse",
            "userId": 1,
          },
        ],
        "requestComplete": true,
        "selectedFromListMenu": "home",
      },
      "setPostData": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "globalReducer": Object {
          "activePage": 1,
          "availablePage": 1,
          "chunkPerPage": 10,
          "groupsPostDataByPage": Array [
            Array [
              Object {
                "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
                "id": 1,
                "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
                "userId": 1,
              },
              Object {
                "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
                "id": 2,
                "title": "qui est esse",
                "userId": 1,
              },
              Object {
                "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
                "id": 2,
                "title": "qui est esse",
                "userId": 1,
              },
              Object {
                "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
                "id": 2,
                "title": "qui est esse",
                "userId": 1,
              },
              Object {
                "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
                "id": 2,
                "title": "qui est esse",
                "userId": 1,
              },
              Object {
                "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
                "id": 2,
                "title": "qui est esse",
                "userId": 1,
              },
              Object {
                "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
                "id": 2,
                "title": "qui est esse",
                "userId": 1,
              },
              Object {
                "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
                "id": 2,
                "title": "qui est esse",
                "userId": 1,
              },
              Object {
                "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
                "id": 2,
                "title": "qui est esse",
                "userId": 1,
              },
              Object {
                "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
                "id": 2,
                "title": "qui est esse",
                "userId": 1,
              },
            ],
          ],
          "inputNameValue": "myname",
          "postData": Array [
            Object {
              "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
              "id": 1,
              "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
            Object {
              "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
              "id": 2,
              "title": "qui est esse",
              "userId": 1,
            },
          ],
          "requestComplete": true,
          "selectedFromListMenu": "home",
        },
        "setPostData": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
